index.js

```diff
@@ -69,74 +87,96 @@ module.exports = {
     if (this.shouldIncludePolyfill()) {
       this.importPolyfill(app);
     }
-  }
-};
+  },
 
-function getAddonOptions(addonContext) {
-  var baseOptions = (addonContext.parent && addonContext.parent.options) || (addonContext.app && addonContext.app.options);
-  return baseOptions && baseOptions.babel || {};
-}
-
-function getBabelOptions(addonContext) {
-  var options = clone(getAddonOptions(addonContext));
-  var ui = addonContext.ui;
-
-  // pass a console object that wraps the addon's `UI` object
-  options.console = {
-    log: function(message) {
-      // fallback needed for support of ember-cli < 2.2.0
-      if (ui.writeInfoLine) {
-        ui.writeInfoLine(message);
-      } else {
-        ui.writeLine(message, 'INFO');
-      }
-    },
-
-    warn: function(message) {
-      // fallback needed for support of ember-cli < 2.2.0
-      if (ui.writeWarnLine) {
-        ui.writeWarnLine(message);
-      } else {
-        ui.writeLine(message, 'WARN');
-      }
-    },
-
-    error: function(message) {
-      // fallback needed for support of ember-cli < 2.2.0
-      if (ui.writeError) {
-        ui.writeError(message);
-      } else {
-        ui.writeLine(message, 'ERROR');
+  _getAddonOptions: function() {
+    return (this.parent && this.parent.options) || (this.app && this.app.options) || {};
+  },
+
+  _shouldCompileModules: function(addonOptions) {
+    var babelOptions = addonOptions.babel;
+    var customOptions = addonOptions['ember-cli-babel'];
+
+    if (customOptions && 'compileModules' in customOptions) {
+      return customOptions.compileModules === true;
+
+    } else if (babelOptions && 'compileModules' in babelOptions) {
+      if (this._shouldShowBabelDeprecations && !this._modulesDeprecationPrinted) {
+        this._modulesDeprecationPrinted = true;
+        this.ui.writeDeprecateLine(
+          'Putting the "compileModules" option in "babel" is deprecated, please put it in "ember-cli-babel" instead.');
       }
-    }
-  };
 
-  // Ensure modules aren't compiled unless explicitly set to compile
-  options.blacklist = options.blacklist || ['es6.modules'];
+      return babelOptions.compileModules === true;
+    } else {
+      return false;
+    }
+  },
 
-  // do not enable non-standard transforms
-  if (!('nonStandard' in options)) {
-    options.nonStandard = false;
-  }
+  _getBabelOptions: function() {
+    var addonOptions = this._getAddonOptions();
+    var options = clone(addonOptions.babel || {});
```

Comment:
Can we name `options` a bit more clearly? I had to look at this multiple times to figure out what it was supposed to be.